dbt/redshift/redshift_etl.py

```diff
@@ -0,0 +1,123 @@
+                host=REDSHIFT_HOST,
+                user=REDSHIFT_USER,
+                password=REDSHIFT_PASSWORD,
+                database=REDSHIFT_DATABASE,
+                port=REDSHIFT_PORT,
+                access_key_id=S3_AWS_ACCESS_KEY_ID,
+                secret_access_key=S3_AWS_SECRET_ACCESS_KEY,
+                start_date=ETL_START_DATE,
+                end_date=ETL_END_DATE,
+                dbt_profiles_dir=DBT_PROFILE_LOCATION,
+                dbt_target=DBT_PROFILE_TARGET_NAME):
+
+    start_date = parsely_utils.parse_datetime_arg(start_date).date()
+    end_date = parsely_utils.parse_datetime_arg(end_date).date()
+
+    # This runs the copy_from_s3 for every day between the start_date and end_date
+    for d in rrule.rrule(rrule.DAILY, interval=1, dtstart=start_date, until=end_date):
+        prefix = 'events/'+ d.strftime('%Y/%m/%d')
+        parsely_redshift.copy_from_s3(network=network,
+                                s3_prefix=prefix,
+                                table_name=table_name,
+                                host=host,
+                                user=user,
+                                password=password,
+                                database=database,
+                                port=port,
+                                access_key_id=access_key_id,
+                                secret_access_key=secret_access_key)
+
+    # This runs dbt once all of the new data has been copied into the raw data table
+    dpl_wd = os.path.join(os.getcwd(), 'dbt/redshift/')
+    subprocess.call(dpl_wd + "run_parsely_dpl.sh " + dbt_profiles_dir + ' ' + dbt_target, shell=True, cwd=dpl_wd)
+
+
+def main():
+    parser = parsely_redshift.get_default_parser("Amazon Redshift utilities for Parse.ly")
+    parser.add_argument('--start_date', required=False, default=ETL_START_DATE,
+                        help='The first day to process data from S3 to Redshift in the format YYYY-MM-DD')
+    parser.add_argument('--end_date', required=False, default=ETL_END_DATE,
+                        help='The last day to process data from S3 to Redshift in the format YYYY-MM-DD')
+    parser.add_argument('--dbt_profiles_dir', required=False, default=DBT_PROFILE_LOCATION,
+                        help='The location from root that contains the .dbt/profiles.yml file, example: /home/user/.dbt/')
+    parser.add_argument('--dbt_target', required=False, default=DBT_PROFILE_TARGET_NAME,
+                        help='The target ie. dev, prod, or test to use within the dbt profiles.yml file.')
+    parser.add_argument('--create-table', action='store_true', default=True,
+                        help='Optional: create the Redshift Parse.ly rawdata table because it does not yet exist.')
+    args = parser.parse_args()
+
+    # Reset dbt_profile to any updated settings:
+    settings_migration = migrate_settings()
+    if not settings_migration:
+        logging.warning("Settings not copied to dbt_profiles.yml successfully.")
```

Comment:
Should this condition raise an Exception that stops the job from occurring? What are the consequences of not getting settings copied to that file?

Reply:
added an Exception :) good eye!